src/Database/Beam/MySQL/Extra.hs

```diff
@@ -323,13 +324,14 @@ getConnection = MySQLM (asks go)
       DebugEnv _ c -> c
       ReleaseEnv c -> c
 
-getPkCols :: MySQLConn -> Text -> MySQLM (Vector Text)
-getPkCols conn nam = do
+getPkCols :: MySQLConn -> Text -> Text -> MySQLM (Vector Text)
+getPkCols conn nam schemaNam = do
   let query = Query $
         "SELECT key_column_usage.column_name " <>
         "FROM information_schema.key_column_usage " <>
-        "WHERE table_schema = schema() " <>
-        "AND constraint_name = 'PRIMARY' " <>
+        "WHERE table_schema = '" <>
+        (encodeUtf8 . fromStrict $ schemaNam) <>
+        "' AND constraint_name = 'PRIMARY' " <>
```

Comment:
why constraint_name?

Reply:
This is an existing query, where it does DB call for getting primary key of the table, should we try to remove this dependancy for euler ?